Standards/scs-03XX-v1-standard-roles.md

```diff
@@ -0,0 +1,271 @@
+---
+title: SCS Standard Roles
+type: Standard
+status: Draft
+track: IAM
+---
+
+## Introduction
+
+SCS aims to provide a standardized role model for RBAC roles across all supported OpenStack API services that applies sensible and consistent permission sets based on a set list of roles defined by a standard.
+It is closely guided by the OpenStack defaults.
+
+## Terminology
+
+The following special terms are used throughout this standard document:
+
+| Term | Meaning |
+|---|---|
+| API | Application Programming Interface, often referring to the REST API interfaces provided by OpenStack and related services |
+| CSP | Cloud Service Provider, provider managing the OpenStack infrastructure |
+| IaaS | Infrastructure-as-a-Service |
+| IAM | Identity and Access Management |
+| RBAC | Role-Based Access Control[^1] established by OpenStack Keystone |
+
+[^1]: [OpenStack Documentation: Role-Based Access Control Overview](https://static.opendev.org/docs/patrole/latest/rbac-overview.html)
+
+## Motivation
+
+The permission settings of OpenStack RBAC roles are configured in service-specific deployment configuration files (usually the respective "`policy.yaml`") in a rather static way and have to be carefully managed.
+In contrast to many of OpenStack's IAM and IaaS resources, these settings cannot be changed via its API at runtime.
+For this reason it is important to have a secure and sensible default configuration in SCS clouds that is both intuitive and flexible enough to cover all necessary use cases of permission models desired by CSPs and customers.
+
+## Design Considerations
+
+One key aspect of the design considerations for this standard is to be as close to the native (upstream) OpenStack role model and role definitions as possible to not introduce unnecessary complexity or divergence from OpenStack.
+Meanwhile the standardized roles and permission sets should cover all scenarios and use cases that SCS deems necessary.
+
+Due to the high level of modularity and the large amount of available services for OpenStack clouds, this standard cannot address all possible manifestations of OpenStack clouds.
+This standard will therefore only cover IaaS APIs and services that are classified as either mandatory or supported by the SCS project.
+
+### Scope Enforcement Compatibility
+
+The API policy library used by OpenStack (oslo.policy) introduced two new [configuration options](https://docs.openstack.org/oslo.policy/latest/configuration/#oslo-policy) during the ongoing RBAC rework of OpenStack[^2]:
+
+- `enforce_scope`
+- `enforce_new_defaults`
+
+Using those new defaults and scope-enforcing options [will currently break](https://governance.openstack.org/tc/goals/selected/consistent-and-secure-rbac.html#the-issues-we-are-facing-with-scope-concept) orchestration tooling such as **OpenStack Heat** and Tacker.
+Due to OpenStack Heat being a service supported by the SCS project, those conflicting options cannot be mandated by a SCS standard.
```

Comment:
After reading the whole standard, I can say this is a bad time to try to write a standard for roles for OpenStack APIs due to the slow an sometimes controversial work on secure RBAC upstream. Nevertheless we have to do it and maybe we should reconsider the way we go here:
1. This is based on the list of mandatory and supported services.
2. Only one of the supported services is the reason, that we cannot use the new and ongoing work of secure RBAC <- Is this correct?
3. We have to made adjustments to the old somehow deprecated way (in Barbican and Octavia) which will at some point hopefully in near future outdated, because of 2.
4. Upstream will not go back to the old way, but will only focus on the new secure RBAC


If we reconsider supporting Heat, do we have any other things that would require us to use the old version?
If there would be nothing holding us from using the new defaults, would those just work out of the box (with setting the two config parameters to true)?

I hesistant to to put so much work into something, that is not supported anymore by upstream OpenStack, but is already under re-development. If there would be a more easy way to directly make use of the new default roles, would that also safe CSPs time and headache when adjusting policies?

Reply:
> 2. Only one of the supported services is the reason, that we cannot use the new and ongoing work of secure RBAC <- Is this correct?

Not anymore, see [this](https://github.com/SovereignCloudStack/issues/issues/396#issuecomment-2200097660) and [this](https://github.com/SovereignCloudStack/issues/issues/396#issuecomment-2253052813).

> 3. We have to made adjustments to the old somehow deprecated way (in Barbican and Octavia) which will at some point hopefully in near future outdated, because of 2.

Only Octavia needs special care in this regard.
Barbican already switches to the new role model as soon as `enforce_new_defaults` is enabled[^1].

> 4. Upstream will not go back to the old way, but will only focus on the new secure RBAC

And we should do the same with standard I think.
The standard should not bother supporting some deprecated leftovers. I'd rather postpone the standard a bit until we get an OpenStack release (maybe 2024.2?) where the new SRBAC just works and we can keep the standard really simple and mandate to use the new defaults.

[^1]: https://github.com/openstack/barbican/blob/d99764e6112cedc6ca168a29fb0279dec2d94288/barbican/common/policies/secrets.py#L85-L88